### ISSUE TITLE
What is the accepted way to add columns that are not driven by data?

### ISSUE THREAD

**Issue by sumeetattree** [miguelcobain/ember-yeti-table#25]
For eg: Adding a column for actions to the table. What is the accepted way of doing this? The following table structure (with filtering enabled)

Reproduced here: https://github.com/sumeetattree/yeti-table-filter-exception-demo

```hbs
<input type="search" placeholder="Search..." value={{filterText}} oninput={{action (mut filterText) value="target.value"}}>
<YetiTable @data={{data}} @filter={{filterText}} as |table|>
  <table.header as |header|>
    <header.column @prop="id">Id</header.column>
    <header.column @prop="name">Name</header.column>
    <header.column>Actions</header.column>
  </table.header>
  <table.body as |body item|>
    <body.row as |row|>
      <row.cell>{{item.id}}</row.cell>
      <row.cell>{{item.name}}</row.cell>
      <row.cell>
        <div class="btn-group">
          <button class="btn btn-sm btn-light">View</button>
          <button class="btn btn-sm btn-light">Edit</button>
        </div>
      </row.cell>
    </body.row>
  </table.body>
</YetiTable>
```
Edit: (Forgot to add the data structure)
```js
// component.js
  ...
  data: [{
    "id": 1,
    "name": "Game #1",
    "state": "lost"
  },{
    "id": 2,
    "name": "Game #2",
    "state": "lost"
  }]
```

fails with:
```js
Error: Assertion Failed: The key provided to get must be a string or number, you passed undefined
    at new EmberError (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:46140:23)
    at assert (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:44973:27)
    at Object.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:26089:119)
    at columns.some.c (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:173830:41)
    at Array.some (<anonymous>)
    at data.filter.row (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:173829:33)
    at Array.filter (<anonymous>)
    at filterData (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:173825:17)
    at Object.<anonymous> (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:174791:48)
    at ComputedProperty.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:27244:36)
    at get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:26105:35)
    at Object.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:37545:29)
    at Object.<anonymous> (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:174795:29)
    at ComputedProperty.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:27244:36)
    at get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:26105:35)
    at Object.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:37545:29)
    at Object.pagedData (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:174694:27)
    at ComputedProperty.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:27244:36)
    at get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:26105:35)
    at Object.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:37545:29)
    at Object.processedData (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:174715:25)
    at ComputedProperty.get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:27244:36)
    at get (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:26105:35)
    at RootPropertyReference.compute (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:17699:35)
    at RootPropertyReference.value (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:17620:53)
    at CapturedNamedArguments.value (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:56493:43)
    at processComponentArgs (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:20607:31)
    at CurlyComponentManager.update (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:20886:29)
    at UpdateComponentOpcode.evaluate (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:54372:21)
    at UpdatingVM.execute (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:55954:24)
    at RenderResult.rerender (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:56158:16)
    at RootState.render (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:21108:44)
    at TransactionRunner.runInTransaction (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:25254:47)
    at InteractiveRenderer._renderRoots (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:21337:90)
    at InteractiveRenderer._renderRootsTransaction (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:21368:22)
    at InteractiveRenderer._revalidate (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:21404:18)
    at invokeWithOnError (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:57905:28)
    at Queue.flush (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:57818:25)
    at DeferredActionQueues.flush (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:57970:31)
    at Backburner._end (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:58400:42)
    at Backburner.end (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:58203:18)
    at Backburner._run (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:58437:26)
    at Backburner._join (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:58420:29)
    at Backburner.join (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:58243:25)
    at join (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:48985:23)
    at http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:22133:42
    at flaggedInstrument (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:46271:88)
    at HTMLInputElement.args (http://192.168.0.244:4200/assets/vendor-d0f9c1dc2ad209ef64583eb6bf115b2e.js:22132:59)
```

This is the line that's causing the failure:
https://github.com/miguelcobain/ember-yeti-table/blob/de9939a0eb96f9626c8b6c7cfb4f807524624f71/addon/-private/utils/filtering-utils.js#L51

Because the `prop` for actions column doesn't exist.

One could get around this issue using the regular `<th>` and `<td>` tags to declare the actions column. Is this the recommended way of adding static columns?

**Comment by cah-briangantzler** [comment]
We ran into this and the problem is that filtering is attempting to filter on the non prop column as you noted.

It should probably be fixed to if the prop is undefined, just skip it, but we fixed it by adding @filterable={{false}} to the column with no prop.

**Comment by sumeetattree** [comment]
I did not know about `@filterable={{false}}`. I tried passing `@filter={{false}}` which obviously failed.
Yeah ideally it shouldn't filter if `prop` is missing.

@miguelcobain Can this be documented? It is missing from the documentation.

**Comment by cah-briangantzler** [comment]
@filterable is documented on the [here](https://miguelcobain.github.io/ember-yeti-table/docs/api/components/yeti-table/head/row/column). The thing is column can be under header.column or head.row.column, but its the same component, so its defined only once.

The better thing to do would be to fix the undefined prop error.

**Comment by sumeetattree** [comment]
My bad. I tried using the search feature for `filterable`. It did not yield any results. Thanks for pointing that out.

> The better thing to do would be to fix the undefined prop error.
Agreed.

**Comment by cah-briangantzler** [comment]
Will make a PR for fixing this, as soon as Im able to make a PR, master branch has a weird dependency in it. 

Please leave this open so I remember to fix it.

**Comment by miguelcobain** [comment]
fixed by #28
